languages/zh-tw/filters.genie

```diff
@@ -0,0 +1,147 @@
+    '當' p:the_out_param_Numeric ('比' | '變得 比') x:constant_Numeric ('大' | '多') ('時' | '的 時候') => C.makeFilter($options, p, '>=', x);
+    '當' p:the_out_param_Numeric ('比' | '變得 比') x:constant_Numeric ('小' | '少') ('時' | '的 時候') => C.makeFilter($options, p, '<=', x);
+}
+
+either_filter = {
+    param:out_param_Any ('等於' | '相等於' | '是') v1:constant_Any ('或' | '或是' | '或者是') v2:constant_Any => {
+        // param is a Value.VarRef
+        //console.log('param: ' + param.name);
+        if (!v1.getType().equals(v2.getType()))
+            return null;
+        if (v1.equals(v2)) // can happen with constants (now, 0, 1, etc.)
+            return null;
+        if (v1.isVarRef && v1.constNumber !== undefined && v2.isVarRef && v2.constNumber !== undefined &&
+            v1.constNumber + 1 !== v2.constNumber) // optimization: avoid CONST_X CONST_Y with X + 1 != Y earlier (before the NN catches it)
+            return null;
+        let vtype = v1.getType();
+        if (vtype.isBoolean) // "is equal to true or false" does not make sense
+            return null;
+        if (!$options.allOutParams.has(param.name + '+' + vtype))
+            return null;
+        return new Ast.BooleanExpression.Atom(param.name, 'in_array', Ast.Value.Array([v1, v2]));
+    };
+    param:out_param_Any '不是' v1:constant_Any '也不是' v2:constant_Any => {
+        // param is a Value.VarRef
+        //console.log('param: ' + param.name);
+        if (!v1.getType().equals(v2.getType()))
+            return null;
+        if (v1.equals(v2)) // can happen with constants (now, 0, 1, etc.)
+            return null;
+        if (v1.isVarRef && v1.constNumber !== undefined && v2.isVarRef && v2.constNumber !== undefined &&
+            v1.constNumber + 1 !== v2.constNumber) // optimization: avoid CONST_X CONST_Y with X + 1 != Y earlier (before the NN catches it)
+            return null;
+        let vtype = v1.getType();
+        if (vtype.isBoolean) // "is neither true nor false" does not make sense
+            return null;
+        if (!$options.allOutParams.has(param.name + '+' + vtype))
+            return null;
+        return new Ast.BooleanExpression.Not(new Ast.BooleanExpression.Atom(param.name, 'in_array', Ast.Value.Array([v1, v2])));
+    };
+}
+
+range = {
+    v1:constant_Numeric ('和' | '跟') v2:constant_Numeric '之間'=> {
+        if (!v1.getType().equals(v2.getType()))
+            return null;
+        if (v1.equals(v2)) // can happen with constants (now, 0, 1, etc.)
+            return null;
+        if (v1.isVarRef && v1.constNumber !== undefined && v2.isVarRef && v2.constNumber !== undefined &&
+            v1.constNumber + 1 !== v2.constNumber) // optimization: avoid CONST_X CONST_Y with X + 1 != Y earlier (before the NN catches it)
+            return null;
+        return [v1, v2];
+    };
+    v1:constant_Date ('和' | '跟') v2:constant_Date '之間'=> {
+        if (!v1.getType().equals(v2.getType()))
+            return null;
```

Comment:
This will never trigger, because range_filter only pairs the range to a numeric (number, currency, measure) parameter, not a date parameter.

Also, the `v1.getType().equals( v2.getType())` check is redundant because you're asking for Date constants through the grammar non-terminal.